src/vs/editor/browser/widget/codeEditorWidget.ts

```diff
@@ -1045,7 +1062,12 @@ export class CodeEditorWidget extends Disposable implements editorBrowser.ICodeE
 	}
 
 	public getContribution<T extends editorCommon.IEditorContribution>(id: string): T | null {
-		return <T>(this._contributions.get(id) || null);
+		const entry = this._contributions.get(id);
+		if (!entry) {
+			return null;
+		}
+
+		return (entry instanceof IdleValue ? entry.value : entry) as T;
```

Comment:
Would this return `Disposable.None` incorrectly here for an `IdleValue` contribution which throws in its ctor?

Reply:
Good call. Looks like `IdleValue` handles exceptions so I will make the `IdleValue` throw instead of returning `Disposable.none`